target_chains/ton/contracts/contracts/Wormhole.fc

```diff
@@ -36,34 +36,47 @@ const int UPGRADE_MODULE = 0x436f7265; ;; "Core" in hex
     int expiration_time = cs~load_uint(64);
     ;; slice keys = cs~load_ref().begin_parse();
     cell keys_dict = cs~load_dict();
-    int keys_count = 0;
+    int key_count = 0;
     int key = -1;
     do {
         (key, slice address, int found) = keys_dict.udict_get_next?(8, key);
         if (found) {
-            keys_count += 1;
+            key_count += 1;
         }
     } until (~ found);
 
-    return (expiration_time, keys_dict, keys_count);
+    return (expiration_time, keys_dict, key_count);
 }
 
 (int, cell, int) get_guardian_set_internal(int index) {
     (slice guardian_set, int found?) = guardian_sets.udict_get?(32, index);
     throw_unless(ERROR_GUARDIAN_SET_NOT_FOUND, found?);
-    (int expiration_time, cell keys, int keys_count) = parse_guardian_set(guardian_set);
-    return (expiration_time, keys, keys_count);
+    (int expiration_time, cell keys, int key_count) = parse_guardian_set(guardian_set);
+    return (expiration_time, keys, key_count);
+}
+
+;; store_data stores data in the contract
+() store_data() impure inline_ref {
+    begin_cell()
+        .store_uint(guardian_set_index, 32)
+        .store_dict(guardian_sets)
+        .store_uint(chain_id, 16)
+        .store_uint(governance_chain_id, 16)
+        .store_uint(governance_contract, 256)
+        .store_dict(consumed_governance_actions)
+        .end_cell()
+        .set_data();
 }
 
 ;; load_data populates storage variables using stored data
-() load_data() impure {
+() load_data() impure inline_ref {
```

Comment:
this code will be used multiple times so its more efficient to use `inline_ref` more info [here](https://docs.ton.org/develop/func/functions#inline_ref-specifier)